src/hooks/useWarmupInfo.ts

```diff
@@ -50,6 +53,31 @@ export const useWarmupClaim = () => {
   );
 };
 
+/**
+ * @returns JS Date for warmup completion for connected wallet
+ */
+export const useWarmupDate = () => {
+  const { data: warmupBalance, isSuccess: balanceSuccess } = useWarmupClaim();
+  const { data: secondsToRebase, isSuccess: rebaseSuccess } = useNextRebase();
+  const { data: epoch, isSuccess: epochSuccess } = useEpoch();
+  const isSuccess = rebaseSuccess && balanceSuccess && epochSuccess;
+
+  const parsedSeconds = parseBigNumber(secondsToRebase || BigNumber.from("0"), 0);
+  const epochLengthSeconds = parseBigNumber(epoch?.length.toBigNumber() || BigNumber.from("0"), 0);
+  // how many fully length epochs are remaining?
+  const warmupLength =
+    parseBigNumber(warmupBalance?.expiry || BigNumber.from("0"), 0) -
+    parseBigNumber(epoch?.number.toBigNumber() || BigNumber.from("0"), 0) -
+    1;
```

Comment:
now we take your expiry epoch, subtract the current epoch & multiply times the epoch length for time remaining